runtime/interpreter/value.go

```diff
@@ -639,6 +648,74 @@ func (v BoolValue) Equal(_ *Interpreter, _ LocationRange, other Value) bool {
 	return bool(v) == bool(otherBool)
 }
 
+func (v BoolValue) Less(interpreter *Interpreter, other ComparableValue, locationRange LocationRange) BoolValue {
+	o, ok := other.(BoolValue)
+	if !ok {
+		panic(InvalidOperandsError{
+			Operation: ast.OperationLess,
+			LeftType:  v.StaticType(interpreter),
+			RightType: other.StaticType(interpreter),
+		})
```

Comment:
Here and below in the other new comparison functions for `BoolValue`:

For numbers, we statically allow comparisons of different concrete number types, which is why there are run-time checks in the comparison functions.

However, comparisons of booleans and other types should be statically rejected, and this check here should be an implementation (unexpected) error instead.

Reply:
Makes sense. Updated it to `panic(errors.NewUnreachableError())`.